### PR TITLE
use exit signal for acct idx bg threads

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem

Acct idx spins up background threads for synchronizing with disk.
These threads need to quit when the validator says to quit.
https://discord.com/channels/428295358100013066/560503042458517505/1016840318840221726

#### Summary of Changes

Pass through an atomic bool that the threads will listen to.

Fixes #

### CODE REVIEW

File: core/src/validator.rs [hunk 1]
```diff
@@ -1468,6 +1468,7 @@ fn load_blockstore(
                 .cache_block_meta_sender
                 .as_ref(),
             accounts_update_notifier,
+            exit,
```
Comment (jeffwashington) [inline core/src/validator.rs:1471]: this is where the exit atomic begins life in this new call chain.

File: runtime/src/bucket_map_holder.rs [hunk 2]
```diff
@@ -370,7 +370,7 @@ impl<T: IndexValue> BucketMapHolder<T> {
             }
             throttling_wait_ms = None;
 
-            if exit.load(Ordering::Relaxed) {
+            if exit.iter().any(|exit| exit.load(Ordering::Relaxed)) {
```
Comment (jeffwashington) [inline runtime/src/bucket_map_holder.rs:373]: this is where the atomic is read

File: runtime/src/accounts_index_storage.rs [hunk 3]
```diff
@@ -57,32 +58,38 @@ impl BgThreads {
...
+                    let system_exit_ = Arc::clone(exit);
                     let in_mem_ = in_mem.to_vec();
 
                     // note that using rayon here causes us to exhaust # rayon threads and many tests running in parallel deadlock
                     Builder::new()
                         .name(format!("solIdxFlusher{:02}", idx))
                         .spawn(move || {
-                            storage_.background(exit_, in_mem_, can_advance_age);
+                            storage_.background(
+                                vec![local_exit_, system_exit_],
```
Comment (t-nelson) [inline runtime/src/accounts_index_storage.rs:81]: nit: i'd have expected the exit vec to be constructed above, alongside the other variables that the closure is going to move

**Review by t-nelson — APPROVED** [review]
lgtm. just nit's from me.  thanks!

fwiw, it'd have been easier to review if the test plumbing were a separate commit from the core plumbing.  plumbing changes dominated the diff, so it was a little tricky to be sure i wasn't missing logical changes

**Pull request merged**